ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/toolbar/DiskResourceViewToolbarImpl.java

```diff
@@ -321,11 +319,15 @@ public void onDiskResourceSelectionChanged(DiskResourceSelectionChangedEvent eve
         editFileMi.setEnabled(editFileMiEnabled);
         editCommentsMi.setEnabled(editCommentsMiEnabled);
         editInfoTypeMi.setEnabled(editInfoTypeMiEnabled);
-        copymetadataMi.setEnabled(metadataMiEnabled);
-        savemetadatami.setEnabled(metadataMiEnabled);
-        bulkmetadataMi.setEnabled(metadataMiEnabled && isFolderSelect);
-        doiMi.setEnabled(metadataMiEnabled && isFolderSelect);
-        editmetadataMi.setEnabled(metadataMiEnabled);
+
+        copymetadataMi.setEnabled(metadataMiEnabled && isReadable(selectedDiskResources.get(0)));
+        savemetadatami.setEnabled(metadataMiEnabled && isReadable(selectedDiskResources.get(0)) );
+        bulkmetadataMi.setEnabled(
+                metadataMiEnabled && isFolderSelect && (isOwner || isWritable(selectedDiskResources)));
+        selectmetadataMi.setEnabled(
+                metadataMiEnabled && isFolderSelect && (isOwner || isWritable(selectedDiskResources)));
+        doiMi.setEnabled(metadataMiEnabled && isFolderSelect && isOwner);
+        editmetadataMi.setEnabled(metadataMiEnabled && isReadable(selectedDiskResources.get(0)));
```

Comment:
I would refactor these `isReadable(selectedDiskResources.get(0))` and `isWritable(selectedDiskResources)` calls into their own booleans, similar to the `isOwner` boolean. I think it makes the code more readable.
While you're at it, the `selectedDiskResources.get(0)` calls could also be refactored into a variable.